src/ApplicationInsights.ServiceFabric.Native/Net45/DependencyTrackingModule/CorrelationIdLookupHelper.cs

```diff
@@ -0,0 +1,324 @@
+        /// Retrieves the app id given the instrumentation key.
+        /// </summary>
+        /// <param name="instrumentationKey">Instrumentation key for which app id is to be retrieved.</param>
+        /// <returns>App id.</returns>
+        private async Task<string> FetchAppIdFromService(string instrumentationKey)
+        {
+            try
+            {
+                SdkInternalOperationsMonitor.Enter();
+
+                Uri appIdEndpoint = this.GetAppIdEndPointUri(instrumentationKey);
+                WebRequest request = WebRequest.Create(appIdEndpoint);
+                request.Method = "GET";
+
+                using (HttpWebResponse response = (HttpWebResponse)await request.GetResponseAsync().ConfigureAwait(false))
+                using (StreamReader reader = new StreamReader(response.GetResponseStream()))
+                {
+                    return await reader.ReadToEndAsync();
+                }
+            }
+            finally
+            {
+                SdkInternalOperationsMonitor.Exit();
+            }
+        }
+
+        /// <summary>
+        /// Strips off any relative path at the end of the base URI and then appends the known relative path to get the app id uri.
+        /// </summary>
+        /// <param name="instrumentationKey">AI resource's instrumentation key.</param>
+        /// <returns>Computed Uri.</returns>
+        private Uri GetAppIdEndPointUri(string instrumentationKey)
+        {
+            return new Uri(this.endpointAddress, string.Format(CultureInfo.InvariantCulture, AppIdQueryApiRelativeUriFormat, instrumentationKey));
+        }
+
+        /// <summary>
+        /// Registers failure for further action in future.
+        /// </summary>
+        /// <param name="instrumentationKey">Instrumentation key for which the failure occurred.</param>
+        /// <param name="ex">Exception indicating failure.</param>
+        private void RegisterFailure(string instrumentationKey, Exception ex)
+        {
+            var ae = ex as AggregateException;
+
+            if (ae != null)
+            {
+                ae = ae.Flatten();
+                if (ae.InnerException != null)
+                {
+                    this.RegisterFailure(instrumentationKey, ae.InnerException);
+                    return;
+                }
+            }
+
+            var webException = ex as WebException;
+            if (webException != null)
+            {
+                this.failingInstrumenationKeys[instrumentationKey] = new FailedResult(
+                    DateTime.UtcNow,
+                    ((HttpWebResponse)webException.Response).StatusCode);
+            }
+            else
+            {
+                this.failingInstrumenationKeys[instrumentationKey] = new FailedResult(DateTime.UtcNow);
+            }
+
+            ServiceFabricSDKEventSource.Log.FetchAppIdFailed(this.GetExceptionDetailString(ex));
+        }
+
+        private string GetExceptionDetailString(Exception ex)
```

Comment:
Didn't you add a utility method for this?

Reply:
fixed.